models/DataObject/ClassDefinition/Helper/Dao.php

```diff
@@ -152,9 +163,13 @@ protected function decryptTables(DataObject\ClassDefinition $classDefinition, ar
         }
     }
 
+    /**
+     * @param string[] $columnsToRemove
+     * @param string[] $protectedColumns
+     */
     protected function removeIndices(string $table, array $columnsToRemove, array $protectedColumns): void
     {
-        if (is_array($columnsToRemove) && count($columnsToRemove) > 0) {
+        if (count($columnsToRemove) > 0) {
```

Comment:
sonarcloud suggests
```suggestion
        if (!empty($columnsToRemove)) {
```

Reply:
https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/

Reply:
then in this case this should be enough
```suggestion
        if ($columnsToRemove) {
```

Reply:
Done https://github.com/pimcore/pimcore/pull/14799/commits/28ec547b21aa9d26b32d14f2024589e47637831b